### ISSUE TITLE
zip files in addons dir is corrupting the install

### ISSUE THREAD

**Issue by nyllewin** [ogri-la/strongbox#424]
**Describe the bug**
downloading addons / updating them via this tool without checking the delete zip will leave them in the interface/addons directory, once wow is closed and battle net is checking or on opening it will fail and get stuck in a ' repair ' 'launch this app as admin' mode.

this should simply not be the case, even storing the addon zips in the interface/addons dir is suboptimal at best. 

**To Reproduce**
*Steps to reproduce the behavior. For example:*
see above

**Expected behavior**
see above

*If applicable, add screenshots to help explain your problem.*

**Additional context**
arch btw lutris wine ge 8.26 bnet

**Comment by torkus** [comment]
> once wow is closed and battle net is checking or on opening it will fail and get stuck in a ' repair ' 'launch this app as admin' mode.

This sounds unrelated to strongbox. Is battle.net *specifically* complaining about zip files in an addon directories?

**Comment by nyllewin** [comment]
> > once wow is closed and battle net is checking or on opening it will fail and get stuck in a ' repair ' 'launch this app as admin' mode.
> 
> This sounds unrelated to strongbox. Is battle.net _specifically_ complaining about zip files in addon directories?

No but this only starts happening after I used strongbox and fixed it by .. redownloading game then using strongbox again without knowing about the zip files.
So I retraced my steps and deleted the recently created addon I put in the addons folder.
at this point I was far enough in to it that I would just copy paste over the WTF and interface folder from previous installs instead of redownloading whole game.
next I noticed even with my new addon deleted it kept happening so I retraced further and thats how I landed here, the point where I used strongbox for the first time to update addons and I then checked the addons directory further only to find the remaining zip files. These were deleted and from that point on I had 0 problems.
So yes, I would say this is very much strongbox related, if not strongbox bug because I realize that its silly of bnet to act up like that then at least it's the cause of the bug with default settings.
It would be nice to have a dedicated storage location for addon zips if users decide to keep them or .. just don't bother thinking about what could happen when zip files are in addons folder for a windows app running on linux. just because the default settings are not taking every possible scenario in to consideration and end up wasting time / bandwidth.

Obviously battle net is not complaining specifically about zip files in addons directory, if that would be the case the app would actually be useful and good but yea thats not the case and never will be.

I also have other files from my windows backup of these folders in there, rar files. never had a problem.

This is all I can say

**Comment by torkus** [comment]
Thanks for the details, @nyllewin , I appreciate you taking the time to elaborate.

> Launch game, fine.
Copy over interface and wtf folder, fine.
launch game, fine.
close game.
battle net suddenly on a mission to secure missile launch codes and ' play ' is now ' update '.
click update - bug happens.
close bnet
delete interface/addons and wtf folder.
open bnet, copy over stuff again with the zips in addons folder.
play, close game, bnet stuck in update bug cycle.

If I'm reading you correctly, it sounds like you can play the game fine with the addons you copied over (including addon zip files) from a previous install, but it's not until after you close WoW that battle.net starts trying to update. Is this correct?

The Interface addons directory also contains official Blizzard addons. I wouldn't be surprised if those were being checked and repaired.

I've had this problem in the past of being in a scanning and repairing loop. It magically resolved itself or something, I don't remember, this was years ago. Doing a quick search I can see that:

1. it's still happening all the time.
2. it's happening for other Blizzard games managed.
3. support recommends all kinds of stuff like deleting `.idx` files, uninstalling and reinstalling, repairing windows, disabling addon managers, etc
4. there are no details as to what is actually being checked during the scan and repaired

We have a situation where nobody knows what is wrong and everybody is just plain guessing at the cause.

But that said, it sounds like you're able to recreate *something*.

If you're up for it:

1. ensure an absolutely clean, stock installation of WoW opens and runs and closes and behaves normally. Open and close a few times, keep an eye on battle.net and confirm for yourself that everything is running smoothly. 
2. copy the WTF and Interface directories of this smoothly running WoW installation to a separate directory called `backup` or similar. This is our known-good working control data. If our experiment fails we should be able to copy it back and it should return to normal.
3. prepare a directory called `test` or similar that contains your previous WTF settings and your previous Interface directory *but with **just** the official Blizzard addons.* No zip files, no other addons, just the official addons.
4. copy the contents of the `test` directory to the right places in your WoW installation
5. start the game, confirm it's working correctly, close the game.

Does battle.net start acting up?

If yes, the problem is either your WTF dir or the old official Blizzard addons.

If not, add a `.zip` file to Interface in your `test` directory and repeat steps 4 + 5.

Does battle.net start acting up?

If yes, the problem is the new variable, the presence of the addon zip file.

If not, unzip the addon file in your `test` directory, then repeat steps 4 + 5 again.

Etc.

I've honestly never encountered this or suspected addon zip files were causing problems in my entire time playing and nobody has ever reported it. @layday , have you ever heard of this happening?

I do [plan on changing on where data is stored](https://github.com/ogri-la/strongbox/blob/aa33f2a6a3e8871fa7944f92c7a631970b6f5e37/TODO.md?plain=1#L265-L268) but it won't be happening in the immediate future.

**Comment by nyllewin** [comment]
> If I'm reading you correctly, it sounds like you can play the game fine with the addons you copied over (including addon zip files) from a previous install, but it's not until after you close WoW that battle.net starts trying to update. Is this correct?

half correct, the part about addon zip files is not correct, other random files .rar files such as rar'd addons from 2018, 2020 are fine but there is not a single zip file in /addons/ or any subdir.

> We have a situation where nobody knows what is wrong and everybody is just plain guessing at the cause.

Yep this is also partially the reason why I'm not making a thread there about something, its just a 0 feed echo chamber where nothing ever gets fully resolves, just somehow fixed and ppl move on.  Plus I would just get ' install on real windows and report back '.


> If you're up for it:

yes, not too soon though as my internet speed is 25Mbit/s at best so, .. yea. But I'll report back

**Comment by layday** [comment]
> I've honestly never encountered this or suspected addon zip files were causing problems in my entire time playing and nobody has ever reported it. @layday , have you ever heard of this happening? It could be new behaviour.

I haven't and I can't reproduce it locally. I assume the Battle.net app keeps logs somewhere, they might shed some light on whatever's going on.

**Comment by torkus** [comment]
I'm closing this ticket but should anyone else suspect strongbox is corrupting installations then please chime in and offer your own experience.

Thanks for opening the ticket @nyllewin , I hope your setup is more stable these days
